core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -62,6 +72,31 @@ oppia.directive('supplementalCard', [
             $scope.helpCardHasContinueButton = false;
           };
 
+          $scope.consumeHint = function() {
+            if (!HintManagerService.areAllHintsExhausted()) {
+              var hint = '';
```

Comment:
Not sure why you do this here. Should go in line 82:

    var hint = HintManagerService.consumeHint();